account_invoice_payment_report/models/account_invoice.py

```diff
@@ -0,0 +1,108 @@
+                        if p.debit_move_id in self.move_id.line_ids
+                    ]
+                )
+                amount_currency = sum(
+                    [
+                        p.amount_currency
+                        for p in payment.matched_debit_ids
+                        if p.debit_move_id in self.move_id.line_ids
+                    ]
+                )
+                if payment.matched_debit_ids:
+                    payment_currency_id = (
+                        all(
+                            [
+                                p.currency_id
+                                == payment.matched_debit_ids[0].currency_id
+                                for p in payment.matched_debit_ids
+                            ]
+                        )
+                        and payment.matched_debit_ids[0].currency_id
+                        or False
+                    )
+            elif self.type in ('in_invoice', 'out_refund'):
+                amount = sum(
+                    [
+                        p.amount
+                        for p in payment.matched_credit_ids
+                        if p.credit_move_id in self.move_id.line_ids
+                    ]
+                )
+                amount_currency = sum(
+                    [
+                        p.amount_currency
+                        for p in payment.matched_credit_ids
+                        if p.credit_move_id in self.move_id.line_ids
+                    ]
+                )
+                if payment.matched_credit_ids:
+                    payment_currency_id = (
+                        all(
+                            [
+                                p.currency_id
+                                == payment.matched_credit_ids[0].currency_id
+                                for p in payment.matched_credit_ids
+                            ]
+                        )
+                        and payment.matched_credit_ids[0].currency_id
+                        or False
+                    )
+            # get the payment value in invoice currency
+            if payment_currency_id and payment_currency_id == self.currency_id:
+                amount_to_show = amount_currency
+            else:
+                currency = payment.company_id.currency_id
+                ctx_date = self.date or fields.Date.today()
+                amount_to_show = currency.with_context(date=ctx_date).compute(
+                    amount, self.currency_id
+                )
+            if float_is_zero(
+                amount_to_show, precision_rounding=self.currency_id.rounding
+            ):
+                continue
+            payment_ref = payment.move_id.name
+            if payment.move_id.ref:
+                payment_ref += ' (' + payment.move_id.ref + ')'
+            payment_vals.append(
+                {
+                    'name': payment.name,
+                    'journal_name': payment.journal_id.name,
```

Comment:
in the additional table only amount and date  is used, seems some fields are redundant, maybe  cut it to only needed vals and left a hook?

Reply:
This is a backport from Odoo code as stated in the comment at the top of the function, so I prefer to keep it as it is.